### ISSUE TITLE
Include commit in version information

### ISSUE THREAD

**Issue by jackwilsdon** [beetbox/beets#1962]
So I'm not entirely sure if this is even possible, but it would be nice to be able to see the current commit in the version information for beets.

This would help with debugging issues when someone is on a new version (i.e. a non-final version), as we can see the exact code that they are running.

The only way I can think of doing this is somehow including it in the `setup.py` so that it is used when installing beets, but as I previously said I'm not entirely sure if this is possible.

Here's the kind of output I'm hoping for:

```
$ beet version
beets version 1.3.18 698203e
```

This wouldn't need to be included for "full" releases as there are already tags for those.

**Comment by sampsyo** [comment]
Sure; this is a good idea! I also don't know how to accomplish it—we should be careful to ensure that the addition doesn't disrupt uncommon circumstances, like when running on Windows or without git installed.

**Comment by jackwilsdon** [comment]
So I found [this code](https://github.com/warner/python-ecdsa/blob/9e21c3388cc98ba90877a1e4dbc2aaf66c67d365/setup.py#L33) that may do it, I'll take a look at implementing this in a bit.

**Comment by sampsyo** [comment]
That looks legit, but I'm a little nervous that it would make it easy to accidentally commit the updated version number. If you type `pip install -e .` inside a git checkout, for example, and then `git commit -a`, you'd clutter up the repo with the current commit hash.

**Comment by jackwilsdon** [comment]
Yeah, what about ignoring it via git? We could check if it exists at runtime and if not, set the commit to unknown?

**Comment by sampsyo** [comment]
Yes, that would work -- check for the separate file's existence with an `except ImportError:`. I still see a couple of problems:

* It's a little bit sad that we'll have to pay the cost of checking for a missing file every time in "official" (non-git) releases, but that's not a big deal.
* How exactly would we include it in a `pip install` from git but not in an `sdist`? We could ignore it in `MANIFEST.in`, but then I think a pre-release install wouldn't include it either.

**Comment by jackwilsdon** [comment]
It looks like we can ["prune" the version file from an `sdist` using the `MANIFEST.in` template](https://docs.python.org/2/distutils/sourcedist.html#principle).
